dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/doi/DOIBackgroundReservationService.java

```diff
@@ -0,0 +1,111 @@
+package edu.harvard.iq.dataverse.doi;
+
+import edu.harvard.iq.dataverse.DatasetDao;
+import edu.harvard.iq.dataverse.globalid.DOIDataCiteServiceBean;
+import edu.harvard.iq.dataverse.persistence.GlobalId;
+import edu.harvard.iq.dataverse.persistence.dataset.Dataset;
+import edu.harvard.iq.dataverse.persistence.dataset.DatasetRepository;
+import edu.harvard.iq.dataverse.settings.SettingsServiceBean;
+import io.vavr.control.Option;
+import io.vavr.control.Try;
+import org.apache.commons.lang.math.NumberUtils;
+
+import javax.annotation.PostConstruct;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+import javax.inject.Inject;
+import java.sql.Timestamp;
+import java.util.Date;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Class designed to reserve doi's in the background.
+ */
+@Startup
+@Singleton
+class DOIBackgroundReservationService {
+
+    private static final Logger logger = Logger.getLogger(DOIBackgroundReservationService.class.getCanonicalName());
+
+    private SettingsServiceBean settingsServiceBean;
+    private DatasetRepository datasetRepository;
+    private DatasetDao datasetDao;
+    private DOIDataCiteServiceBean doiDataCiteService;
+
+    private final Timer timer = new Timer();
```

Comment:
Is there a particular reason that we don't use Timers from jee?

Reply:
Yes, I tried using TimerService but if the method would've throw exception there was no way to stop timer destruction (the method would be executed twice and then time would die). Since it is under JEE control I can't wrap it around try{} so I didn't see any way to prevent it. There were some workarounds on stackoverflow but they were not satisfying and this is cleaner way.